File: render_liquid.rb
#!/usr/bin/env ruby
require 'json'
require 'liquid'


def render_liquid
    template = Liquid::Template.parse(ARGV[0], :error_mode => :strict)
    context = JSON.parse(ARGV[1])
    return template.render(JSON.parse(ARGV[1]), { strict_variables: true, strict_filters: true })
end


if __FILE__ == $0
    print render_liquid()
end


# TODO: reuse processes via pipes, Sinatra, or xml-rpc
# https://ruby-doc.org/stdlib-2.3.1/libdoc/xmlrpc/rdoc/XMLRPC/Client.html
# (or just stick it in AWS Lambda)

File: render_liquid.py
#!/usr/bin/env python
import json
import subprocess


def render_liquid(template, context):
    # TODO: timeout, better error handling
    # TODO: reuse processes via pipes, Sinatra, or xml-rpc
    try:
        result = subprocess.run(
            ['./render_liquid.rb', template, json.dumps(context)],
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            check=True,
        )
    except:
        import pdb; pdb.set_trace()
    return result.stdout.decode('utf-8')


if __name__ == '__main__':
    import sys
    print(render_liquid(sys.argv[1], sys.argv[2]), end='')
